Update benchmark test to latest changes of subscription logic

GrantSeltzer opened the pull request:
This updates `test/benchmark` to the latest state of the repository

Signed-off-by: grant <grant@capsule8.com>

orangematt commented on test/benchmark/main.go:
```diff
@@ -52,6 +58,21 @@ func init() {
+	manager = services.NewServiceManager()
+
+	if len(sensorConfig.Sensor.ListenAddr) > 0 {
+		testSensor, err := sensor.NewSensor()
```
I'm suspicious of this. I do not think that it does what you think it does. Here, the global `testSensor` does not receive the assignment, but instead a new local `testSensor` that shadows the global gets created. `go vet -shadow` should warn about this. I haven't run the code here, but I whipped up a quick test to verify that Go's behavior with `:=` is what I suggest. It is. Even if I am wrong about this, it should be made clearer; the intent is clear, but what actually happens is ambiguous at best.

ddz replied:
I was wondering why `go vet` didn't flag this in CI and it turns out that it runs:
`go vet -shadow ./cmd/... ./pkg/... ./examples/...`

We should add `./test` to the list of directories that get checked here.

GrantSeltzer replied:
Ah,  nice catch @orangematt, I fixed this and confirmed it to work how I intended. 

@ddz I will add `./test` to the list of checked directories.

orangematt commented on Makefile:
```diff
@@ -61,7 +61,7 @@ BINS=$(patsubst %,bin/%,$(CMDS)) \
 
 # All source directories that need to be checked, compiled, tested, etc.
-SRC=./cmd/... ./pkg/... ./examples/...
+SRC=./cmd/... ./pkg/... ./examples/... ./test/benchmark/...
```
Why not `./test/...`?

GrantSeltzer replied:
The functional tests don't get built in the same way as the rest of the source code

orangematt replied:
Makes sense not adding to `SRC`. It would be nice for `gofmt`, `go vet`, and `golint` to run on those sources, though. Is there any reason that cannot be done in the check rule? Maybe add `CHECKSRC=$(SRC) ./test/...` and then use `CHECKSRC` in the check rule?

GrantSeltzer replied:
Good call. I just did that which exposed some vet issues with a functional test. Going to fix those as part of this along with adding CHECKSRC as you suggested.

GrantSeltzer commented:
@orangematt 

I updated the Makefile to run checks on `test/...` and fixed all lint+vet issues in functional tests (as well as fixing another outdated test)
